.github/workflows/publish-release-image.yaml

```diff
@@ -0,0 +1,112 @@
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: https://ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Download wheels
+        uses: actions/download-artifact@v4
+        with:
+          name: eager-dist-${{ matrix.os }}-${{ matrix.arch }}
+      - name: Get the name of the wheel and set up env variables
+        id: generate-tag-name
+        run: |
+          echo "WHEEL_FILENAME=$(ls -1 *.whl)" >> $GITHUB_ENV
+          REPO_IMAGE_NAME=ghcr.io/${{ github.repository }}/tt-metalium-${{ matrix.os }}-amd64-dev/${{ matrix.arch }}
+          echo "REPO_IMAGE_NAME=$REPO_IMAGE_NAME" >> $GITHUB_ENV
+          TAG_NAME=$REPO_IMAGE_NAME:${{ inputs.version }}
+          echo "TAG_NAME=$TAG_NAME" >> $GITHUB_ENV
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          push: true
+          build-args: |
+            WHEEL_FILENAME=${{ env.WHEEL_FILENAME }}
+            BASE_IMAGE_NAME=tt-metalium/${{ matrix.os }}-amd64
+          tags: ${{ env.TAG_NAME }}
+          context: .
+          file: dockerfile/release.Dockerfile
+  smoke-test-docker-image:
+    needs: create-docker-image
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+        test_group:
+          [
+              {
+                arch: grayskull,
+                runs-on: ["cloud-virtual-machine", "E150", "in-service"],
+                cmd: pytest tests/end_to_end_tests,
+              },
+              {
+                arch: wormhole_b0,
+                runs-on: ["cloud-virtual-machine", "N150", "in-service"],
+                cmd: pytest tests/end_to_end_tests,
+              }
+          ]
+    env:
+      TT_METAL_ENV: ${{ vars.TT_METAL_ENV }}
+      ARCH_NAME: ${{ matrix.test_group.arch }}
+      LOGURU_LEVEL: INFO
+      LD_LIBRARY_PATH: ${{ github.workspace }}/build/lib
+    runs-on: ${{ matrix.test_group.runs-on }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Run smoke test on the image
+        timeout-minutes: ${{ inputs.timeout }}
+        uses: ./.github/actions/docker-run
+        with:
+          docker_os_arch: tt-metalium-${{ matrix.os }}-amd64-release/${{ matrix.test_group.arch }}
+          docker_password: ${{ secrets.GITHUB_TOKEN }}
+          run_args: |
+            ${{ matrix.test_group.cmd }}
+  tag-docker-image-as-latest:
+    needs: [smoke-test-docker-image, create-docker-image]
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+        arch: [grayskull, wormhole_b0]
+    runs-on:
+      - build-docker
+      - in-service
+    steps:
+      - name: Tag latest if this is a major version release
+        if: ${{ inputs.is_major_version }}
+        run: |
+          set -eu # basic shell hygiene
+          REPO_IMAGE_NAME=ghcr.io/${{ github.repository }}/tt-metalium-${{ matrix.os }}-amd64-release/${{ matrix.arch }}
```

Comment:
* Hmm... we did we drop the `tt-metalium? It's inconsistent and I'm not a fan

  In `build-docker-artifact`: `ghcr.io/${{ github.repository }}/tt-metalium/ubuntu-22.04-amd64:latest`
  but here: `ghcr.io/${{ github.repository }}/tt-metalium-ubuntu-22.04-amd64-release/${{ matrix.arch }}`

  I like the former style better, so how about 

  `ghcr.io/${{ github.repository }}/tt-metalium/ubuntu-22.04-amd64-release:latest-${{ matrix.arch }}` ?

  Since the wheels are arch-dependent. 

* Also, I'm wondering if we should get `amd64` from a `host_arch` matrix instead. We're letting it clutter everywhere a bit too much.

Reply:
This is the style @TT-billteng proposed in this ticket https://github.com/tenstorrent/tt-metal/issues/12495

Importantly, other teams are using the same style i.e. take a look at 

https://github.com/orgs/tenstorrent/packages

<img width="1103" alt="Screenshot 2024-12-03 at 11 48 52 AM" src="https://github.com/user-attachments/assets/40c8ff80-e2eb-4332-930e-14f52c4caaad">

Reply:
OK - fine

Reply:
It seems like we aren't matching the format with other teams there too.
Do we need to go with the '-' route?
not that its important but better to fix this now while we can

Reply:
@ttmchiou Could you highlight the discrepancy? 
`tt-metalium-ubuntu-20.04-amd64-release` vs `tt-mlir-ubuntu-20-04` 
Is that the dots within the Ubuntu version? 

I also notice that some of the images have the {base, release, dev} in the middle of the image. 

At least we are going by Bill's proposed changes - #12495